docs/_layouts/global.html

```diff
@@ -207,14 +205,79 @@ <h1 class="title">{{ page.title }}</h1>
     })
 </script>
 
+<!-- Pure JavaScript implementation of ScrollSpy -->
 <script>
-    var controller = new ScrollMagic.Controller();
+    //finds toc and automatically sorts by main list items as parents and nested list items as children
 
-    new ScrollMagic.Scene({
-        offset: 20
-    })
-    .setClassToggle(".navbar-inner", "shrunk")
-    .addTo(controller);
+    var toc = document.getElementById("toc");
+    if(toc != null) {
+        var topPadding = 12; // amount of space above header to start noticing a section
+        var navBar = document.getElementById("topbar").offsetHeight;
+        var tocEl = toc.nextElementSibling.getElementsByTagName("li");
+        var sections = toc.nextElementSibling.getElementsByTagName("a");
+        sections = Array.prototype.slice.call(sections);
+        var sectionsLen = sections.length;
+        var parentIndex;
+        var parentTag = sections[0].getAttribute("href") == "#" ? "H1" : document.getElementById(sections[0].getAttribute("href").substring(1)).nodeName;
```

Comment:
In general, could you add more comments in the javascript describing some of the more involved operations?